docs/src/tutorial.rst

```diff
@@ -0,0 +1,505 @@
+
+              // Create some dummy data to populate object
+              const now = new Date();
+              var data = name + " created: " + now;
+
+              var base64data = new Buffer(data, 'binary');
+
+              S3.putObject({
+                Bucket: bucketName,
+                Key: name,
+                Body: base64data,
+                ContentType: 'application/json'
+              }).promise()
+                .then(function() {
+                  var response = makeResponse(event.apples);
+                  console.log("POST /name response: " + response);
+                  callback(null, response);
+                })
+                .catch(rejectedPromise);
+            }
+          }
+          else if (event.httpMethod == "DELETE") {
+            // DELETE /name
+            // Make sure we have a name
+            var pathLength = event.path.length
+
+            if (pathLength == 1) {
+              // Return error
+              var body = "NAME arg missing (/ supplied, not /name)";
+              var response = makeError(body);
+              callback(null, response);
+            }
+            else {
+              name = getName(event.path);
+
+              S3.deleteObject({ Bucket: bucketName, Key: name })
+                .promise()
+                .then(function(data) {
+                  var body = { success: true }
+                  var response = makeResponse(body);
+                  callback(null, response);
+              })
+                .catch(rejectedPromise);
+            }
+          }
+          else {
+            // Return error
+            var body = "HTTP method " + event.httpMethod + " not supported";
+            var response = makeError(body);
+            callback(null, response);
+          }
+        }
+      }
+
+      function rejectedPromise(error) {
+        var body = error.stack || JSON.stringify(error, null, 2);
+        var response = makeError(body);
+        callback(null, response);
+      }
+    }
+
+You probably noticed we also introduced three new helper functions:
+**getName**, **makeResponse**, and **makeError**.
+Add these to *apples.js*.
+
+.. code-block:: js
+
+    // Get name from path /name
+    function getName(path) {
+      var name = path.substring(1, path.length);
+      return name;
+    }
+
+    // Create JSON response from body
+    function makeResponse(responseBody) {
+      var response = {
+        statusCode: 200,
+        headers: {},
+        body: JSON.stringify(responseBody)
+      };
+
+      return response;
+    }
+
+    // For bad requests
+    function makeError(body) {
+      var response = {
+        statusCode: 404,
+        headers: {},
+        body: JSON.stringify(body)
+      };
+
+      return response;
+    }
+
+Wire these functions up to our API Gateway code in *apple_service.ts*
+by adding the following code at the end of the constructor.
+
+.. code-block:: ts
+
+    const apple = api.root.addResource('{name}');
```

Comment:
Usually `POST` is done on the parent (in this case `api.root`), with any attributes passed in the application/json request body:

```
POST /apples
{
  "name": "my-new-apple"
}
```

Reply:
You are correct, but the example is so simple, requiring just the name.